src/Umbraco.Infrastructure/Security/MemberRoleStore.cs

```diff
@@ -0,0 +1,342 @@
+                    //TODO: if nothing changed, do we need to report this?
+                    return Task.FromResult(IdentityResult.Success);
+                }
+                else
+                {
+                    //TODO: throw exception when not found, or return failure?
+                    return Task.FromResult(IdentityResult.Failed(_memberGroupNotFoundError));
+                }
+
+            }
+            catch (Exception ex)
+            {
+                return Task.FromResult(IdentityResult.Failed(new IdentityError { Code = ex.Message, Description = ex.Message }));
+            }
+        }
+
+        /// <inheritdoc />
+        public Task<IdentityResult> DeleteAsync(TRole role, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                ThrowIfDisposed();
+
+                if (role == null)
+                {
+                    throw new ArgumentNullException(nameof(role));
+                }
+
+                if (!int.TryParse(role.Id, out int roleId))
+                {
+                    //TODO: what identity error should we return in this case?
+                    return Task.FromResult(IdentityResult.Failed(_intParseError));
+                }
+
+                IMemberGroup memberGroup = _memberGroupService.GetById(roleId);
+                if (memberGroup != null)
+                {
+                    _memberGroupService.Delete(memberGroup);
+                }
+                else
+                {
+                    return Task.FromResult(IdentityResult.Failed(_memberGroupNotFoundError));
+                }
+
+                return Task.FromResult(IdentityResult.Success);
+            }
+            catch (Exception ex)
+            {
+                return Task.FromResult(IdentityResult.Failed(new IdentityError { Code = ex.Message, Description = ex.Message }));
+            }
+        }
+
+        /// <inheritdoc />
+
+        public Task<string> GetRoleIdAsync(TRole role, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            return Task.FromResult(role.Id);
+        }
+
+        /// <inheritdoc />
+        public Task<string> GetRoleNameAsync(TRole role, CancellationToken cancellationToken = default)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            ThrowIfDisposed();
+
+            if (role == null)
+            {
+                throw new ArgumentNullException(nameof(role));
+            }
+
+            return Task.FromResult(role.Name);
+        }
+
+        /// <inheritdoc />
+        public Task SetRoleNameAsync(TRole role, string roleName, CancellationToken cancellationToken = default)
```

Comment:
https://github.com/dotnet/aspnetcore/blob/main/src/Identity/Extensions.Stores/src/RoleStoreBase.cs#L116

Reply:
Like: ` role.Name = roleName;`?

@scottbrady91 thanks, just to check: 

Does that mean any role name change that needs to be persisted via the `MemberGroupService` will need to be done via  `_roleManager.UpdateAsync(role);`? 

What if the user only wants to change the role name and persist it to the Umbraco store?
`_roleManager.SetRoleNameAsync(role);`

Reply:
Yep like @scottbrady91  mentions, these methods will just need to set properties and then they will be persisted with UpdateAsync. This means we need to do change tracking on all the properties (if we aren't already) just like we do with the MemberIdentity. And yes if using this method, you then need to call UpdateAsync.

Is there a reason we aren't inheriting from RoleStoreBase? Seems like we would get most of the functionality for free? and don't have to implement methods like this.

Reply:
Ah, I see RoleStoreBase is a large object that implements much more than we will support. Make sense we just implement the interface 👍  I'll fix this one up so that it's just setting the property and it will be persisted during update.